src/pages/iou/request/step/IOURequestStepWaypoint.tsx

```diff
@@ -13,92 +13,89 @@ import InputWrapperWithRef from '@components/Form/InputWrapper';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type {Route as Routes} from '@src/ROUTES';
 import ROUTES from '@src/ROUTES';
-import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import type * as OnyxTypes from '@src/types/onyx';
+import type {Waypoint} from '@src/types/onyx/Transaction';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
 import withWritableReportOrNotFound from './withWritableReportOrNotFound';
 
-const propTypes = {
-    /** Navigation route context info provided by react navigation */
-    route: IOURequestStepRoutePropTypes.isRequired,
+type Location = {
+    lat?: number;
+    lng?: number;
+};
 
-    /* Onyx props */
-    /** The optimistic transaction for this request */
-    transaction: transactionPropTypes,
+type Geometry = {
+    location: Location;
+};
 
-    /* Current location coordinates of the user */
-    userLocation: PropTypes.shape({
-        /** Latitude of the location */
-        latitude: PropTypes.number,
+type WaypointValues = Record<string, string>;
 
-        /** Longitude of the location */
-        longitude: PropTypes.number,
-    }),
+type MappedWaypoint = {
+    /** Waypoint name */
+    name?: string;
 
-    /** Recent waypoints that the user has selected */
-    recentWaypoints: PropTypes.arrayOf(
-        PropTypes.shape({
-            /** The name of the location */
-            name: PropTypes.string,
+    /** Waypoint description */
+    description: string;
 
-            /** A description of the location (usually the address) */
-            description: PropTypes.string,
+    /** Waypoint geometry object cointaing coordinates */
+    geometry: Geometry;
+};
 
-            /** Data required by the google auto complete plugin to know where to put the markers on the map */
-            geometry: PropTypes.shape({
-                /** Data about the location */
-                location: PropTypes.shape({
-                    /** Latitude of the location */
-                    lat: PropTypes.number,
+type IOURequestStepWaypointOnyxProps = {
+    /** List of recent waypoints */
+    recentWaypoints: OnyxEntry<MappedWaypoint[]>;
 
-                    /** Longitude of the location */
-                    lng: PropTypes.number,
-                }),
-            }),
-        }),
-    ),
+    userLocation: OnyxEntry<OnyxTypes.UserLocation>;
 };
 
-const defaultProps = {
-    recentWaypoints: [],
-    transaction: {},
-    userLocation: undefined,
-};
+type IOURequestStepWaypointProps = {
+    route: {
+        params: {
+            iouType: ValueOf<typeof CONST.IOU.TYPE>;
+            transactionID: string;
+            reportID: string;
+            backTo: Routes | undefined;
+            action: ValueOf<typeof CONST.IOU.ACTION>;
+            pageIndex: string;
+        };
+    };
+    transaction: OnyxEntry<OnyxTypes.Transaction>;
+} & IOURequestStepWaypointOnyxProps;
```

Comment:
Should we use StackScreenProps here?
```suggestion
type IOURequestStepWaypointProps = IOURequestStepWaypointOnyxProps & StackScreenProps<...>
```

Reply:
Actually no, because StackSceenProps is providing more properties which are passed to this component 
```
<IOURequestStepWaypoint
            // Put the transactionID into the route params so that WaypointEdit behaves the same when creating a new waypoint
            // or editing an existing waypoint.
            route={{
                params: {
                    ...route.params,
                    transactionID,
                },
            }}
        />